samples/ask-akka-agent/src/main/java/akka/ask/agent/application/NewAskAkkaAgent.java

```diff
@@ -8,17 +8,26 @@
 @AgentDescription(name = "Ask Akka", description = "Expert in Akka")
 public class NewAskAkkaAgent extends Agent {
 
+  private final Knowledge knowledge;
+
   private final String sysMessage = """
       You are a very enthusiastic Akka representative who loves to help people!
       Given the following sections from the Akka SDK documentation, answer the question using only that information, outputted in markdown format.
       If you are unsure and the text is not explicitly written in the documentation, say:
       Sorry, I don't know how to help with that.
-      """; // <1>
+      """;
+
+  public NewAskAkkaAgent(Knowledge knowledge) {
+    this.knowledge = knowledge;
+  }
 
   public Agent.Effect<String> ask(String question) {
+    var questionWithKnowledge = knowledge.addKnowledge(question);
+
     return effects()
         .systemMessage(sysMessage)
-        .userMessage(question)
+        .userMessage(questionWithKnowledge)
```

Comment:
and yes, this agent can now answer "What are the core components of Akka?" in the correct way